### PR TITLE
[Bugfix] stale page update in the Calling samples

### PR DESCRIPTION
**Pull request by dmceachernmsft**
# What

Add hook that auto refreshed the page if it is cached.
update docs to show a snippet of how to do this as well.
# Why

On mobile if the user dismisses the page the browser will kill all subscriptions to the page and load it from cache when they open the browser next. this change causes the samples to update.

[BUG 3755439](https://skype.visualstudio.com/SPOOL/_workitems/edit/3755439)
# How Tested

Tested locally and deployed to an app service

### CODE REVIEW

File: packages/storybook/stories/Troubleshooting.stories.mdx [hunk 1]
```diff
@@ -98,3 +98,32 @@ can be caused by passing arguments to our factory functions too often. We recomm
...
+  });
+  return () => {
+    window.removeEventListener('pageshow', () => {
+      window.location.reload();
+    });
+  };
+}, []);
+```
+
+This `useEffect` will add and event listener for the `pageshow` event, this event when triggered
```
Comment (mgamis-msft) [inline packages/storybook/stories/Troubleshooting.stories.mdx:127]: "... will add **an** event listener ..."

File: packages/storybook/stories/Troubleshooting.stories.mdx [hunk 2]
```diff
@@ -98,3 +98,32 @@ can be caused by passing arguments to our factory functions too often. We recomm
...
+will share if the page is persisted or `cached`. If the page is cached we will reload the app so the
```
Comment (mgamis-msft) [inline packages/storybook/stories/Troubleshooting.stories.mdx:128]: nit: why is cached quoted here :)?

File: packages/storybook/stories/Troubleshooting.stories.mdx [hunk 3]
```diff
@@ -98,3 +98,32 @@ can be caused by passing arguments to our factory functions too often. We recomm
 or disposing of the adapter or stateful client before attempting to create a new one.
 
 Another common cause for this error is using strict mode in your application. Please see [React's documentation](https://react.dev/reference/react/StrictMode) on strict mode to learn more.
+
+## Mobile browser refresh
+
+This is an issue when using a mobile browser and the user closes the browser. When they do this
+the browser will load the page cached which all of the subscriptions will be lost. To fix this as the
```
Comment (mgamis-msft) [inline packages/storybook/stories/Troubleshooting.stories.mdx:105]: For the second sentence, do you mean? "When they do this
the browser will load the page but all of the cached subscriptions will be lost"

**Review by mgamis-msft — APPROVED** [review]

File: packages/storybook/stories/Troubleshooting.stories.mdx [hunk 4]
```diff
@@ -98,3 +98,32 @@ can be caused by passing arguments to our factory functions too often. We recomm
 or disposing of the adapter or stateful client before attempting to create a new one.
 
 Another common cause for this error is using strict mode in your application. Please see [React's documentation](https://react.dev/reference/react/StrictMode) on strict mode to learn more.
+
+## Mobile browser refresh
+
+This is an issue when using a mobile browser and the user closes the browser. When they do this
```
Comment (edwardlee-msft) [inline packages/storybook/stories/Troubleshooting.stories.mdx:104]:
```suggestion
This is an issue when using a mobile browser and the user closes the browser. 
```
Do not need the when they do this portion

**Review by edwardlee-msft — APPROVED** [review]

**Pull request merged**